include/boost/geometry/util/math.hpp

```diff
@@ -191,6 +192,66 @@ struct square_root<T, true>
 };
 
 
+
+template
+<
+    typename T,
+    bool IsFundemantal = boost::is_fundamental<T>::value /* false */
+>
+struct modulo
+{
+    typedef T return_type;
+
+    static inline T apply(T const& value1, T const& value2)
+    {
+        // for non-fundamental number types assume that sqrt is
+        // defined either:
+        // 1) at T's scope, or
+        // 2) at global scope, or
+        // 3) in namespace std
+        using ::fmod;
+        using std::fmod;
+
+        return fmod(value1, value2);
+    }
+};
+
+template <typename Fundamental, bool IsIntegral>
```

Comment:
So here IsIntegral could be initialized to boost::is_integral<Fundamental>::value by default.

But actually, I think in general we're specializing math functions for floating-point types. I'm aware that this is done this way in order to support `ttmath::Big` above in the `modulo` struct. So for user-defined types it's required that there should be a function `fmod()`. I think that we should consider the implementation of 3 versions:
- for built-in integrals - using `operator%`
- for built-in floating-point types - calling `fmod()`
- for user-defined types - calling `mod()` because it's more generic name (and maybe additionally a default-version of this function which could be found by the ADL, which'd use operator% or alternativaly call a member function of user-defined type)